### DISCUSSION TITLE
Enable control of `UseShellExecute`

### DISCUSSION THREAD

**Discussion by gitfool** [Tyrrrz/CliWrap#114] [Q&A] [answered]
I'm not sure but this may be out of scope for what `CliWrap` is intended to do... I'd like to shell out to the [vs code cli](https://code.visualstudio.com/docs/editor/command-line#_core-cli-options), wait for a file to be edited, and then continue. From my experiments with [`ProcessStartInfo`](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.processstartinfo) this does not work unless [`UseShellExecute`](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.processstartinfo.useshellexecute#System_Diagnostics_ProcessStartInfo_UseShellExecute) is enabled _and_ redirection is disabled:
```c#
var process = Process.Start(new ProcessStartInfo
{
    FileName = "code.cmd",
    ArgumentList = { "--new-window", "--wait", file },
    UseShellExecute = true
});
if (process != null)
{
    await process.WaitForExitAsync();
    if (process.ExitCode == 0)
    {
        ...
    }
}
```
`CliWrap` is already being used elsewhere in the app so it would be nice to use it here too.

**Comment by Tyrrrz** [comment, marked as answer]
Shell execute unfortunately can't be enabled because that prevents from reading stdout/stderr, which is pretty much the main purpose of using CliWrap. At that point it wouldn't be much different than using `Process` directly, but most importantly allowing shell execute to be configurable will diverge the internal implementation into two separate workflows. So in essence this is out of scope.
